baremetal/metal3machine_manager_test.go

```diff
@@ -4761,22 +4763,29 @@ func newBareMetalHost(name string,
 	status *bmov1alpha1.BareMetalHostStatus,
 	powerOn bool,
 	autoCleanMode string,
-	clusterlabel bool) *bmov1alpha1.BareMetalHost {
+	clusterlabel bool,
+	bmhUID ...string) *bmov1alpha1.BareMetalHost {
 	if name == "" {
 		return &bmov1alpha1.BareMetalHost{
 			ObjectMeta: testObjectMeta("", namespaceName, ""),
 		}
 	}
 
+	BMHUid := types.UID(bmhuid)
+	if len(bmhUID) > 0 {
+		BMHUid = types.UID(bmhUID[0])
+	}
```

Comment:
I'm getting confused with all these BMH UIDs. :flushed: 
Why do we need `bmhuid` that is coming from somewhere outside of this file? Is it like a default if nothing is specified?
Could you add some comment to describe the expected behavior?

Reply:
yes, it is confusing, let me fix it. In `metal3machine_manager_test.go` we were already using it from two places one `bmhuid` was coming from suite_test.go and one `Bmhuid` was in the same file, which needs to be refactored. I will change it to the same uid everywhere in `metal3machine_manager_test.go` and also fix the above function.
And yes aim of the above implementation was to set the default value if nothing is passed.

Reply:
I have done some fixes can you check if they look ok.